packages/nextjs/hooks/scaffold-eth/useScaffoldEventRead.ts

```diff
@@ -0,0 +1,124 @@
+>({
+  contractName,
+  eventName,
+  fromBlock,
+  filters,
+  blockData,
+  transactionData,
+  receiptData,
+}: {
+  contractName: TContractName;
+  eventName: TEventName;
+  fromBlock?: number;
+  filters?: any;
+  blockData?: boolean;
+  transactionData?: boolean;
+  receiptData?: boolean;
+}) => {
+  const [events, setEvents] = useState(<any>[]);
+  const { data: deployedContractData } = useDeployedContractInfo(contractName);
+  const provider = useProvider();
+
+  const contract = useContract({
+    address: deployedContractData?.address,
+    abi: deployedContractData?.abi,
+    signerOrProvider: provider,
+  });
+
+  useEffect(() => {
+    async function readEvents() {
+      if (!contract) {
+        console.log("contract not found");
+        return;
+      }
+      try {
+        if (!fromBlock) {
+          fromBlock = await provider.getBlockNumber() - 1000;
+        }
+
+        const fragment = contract.interface.getEvent(eventName);
+        if (!fragment) {
+          console.log("event not found");
+          return;
+        }
+
+        const emptyIface = new ethers.utils.Interface([])
+        const topicHash = emptyIface.getEventTopic(fragment)
+        const topics = <any>[ topicHash ];
+
+        const indexedParameters = fragment.inputs.filter((input) => input.indexed);
+
+        if (indexedParameters.length > 0 && filters) {
+          const indexedTopics = indexedParameters.map((input) => {
+            const value = filters[input.name];
+            if (value === undefined) {
+              return null;
+            }
+            if (Array.isArray(value)) {
+              return value.map((v) => ethers.utils.hexZeroPad(ethers.utils.hexlify(v), 32));
+            }
+            return ethers.utils.hexZeroPad(ethers.utils.hexlify(value), 32);
+          });
+          topics.push(...indexedTopics);
+        }
+
+        const logs = await provider.getLogs({
+            address: deployedContractData?.address,
+            topics: topics,
+            fromBlock: fromBlock,
+        });
+        const newEvents = [];
+        for (let i=0; i<logs.length; i++) {
+          let block;
+          if (blockData) {
+            block = await provider.getBlock(logs[i].blockHash);
+          }
+          let transaction;
+          if (transactionData) {
+            transaction = await provider.getTransaction(logs[i].transactionHash);
+          }
+          let receipt;
+          if (receiptData) {
+            receipt = await provider.getTransactionReceipt(logs[i].transactionHash);
+          }
+          const log = {
+            log: logs[i],
+            args: contract.interface.parseLog(logs[i]).args,
+            block: block,
+            transaction: transaction,
+            receipt: receipt,
+          }
+          newEvents.push(log);
+        }
+        setEvents(newEvents.reverse());
+      }
+      catch (e) {
+          console.log(e);
+      }
+    }
+    readEvents();
```

Comment:
Only call `readEvents` if `useDeployedContractInfo` is not loading

Reply:
Added!